deployments/scripts/smoke-summoner.sh

```diff
@@ -97,7 +123,7 @@ echo "Starting phase 2 run..."
 cargo run --quiet --release --bin summonerd -- start --phase 2 --storage-dir /tmp/summonerd --fvk $SUMMONER_FVK --node http://127.0.0.1:8080 --bind-addr 127.0.0.1:8082 &
 phase2_pid="$!"
 # If script ends early, ensure phase 2 is halted.
-trap 'kill -9 "$phase2_pid"' EXIT
+trap 'kill -9 "$phase2_pid"' EXIT INT
```

Comment:
Hmm, I left a summoner-smoke running locally, and the pd process never exited. AIUI, each `trap` declaration clobbers the previous one, so I think what we really want is to restate the trap by appending pids to it. That way an exit/interrupt will still clean up the older processes. Not important to massage, given that this script only runs in CI, but mentioning it in case you have another round of clean up before merge.